### PR TITLE
Browser cryptography breaking changes

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #20143.

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/3.1-5.0?branch=pr-en-us-20606#systemsecuritycryptography-apis-not-supported-on-blazor-webassembly).

### CODE REVIEW

**Comment by gewarren** [comment]
It seems like we should also add the PNSE to the API docs for all the affected methods. Or is it just too many?

It would look something like this: https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.rsa.create?view=net-5.0#System_Security_Cryptography_RSA_Create_System_Int32_.

File: docs/core/compatibility/aspnetcore.md [hunk 1]
```diff
@@ -193,6 +194,10 @@ The following breaking changes in ASP.NET Core 3.0, 3.1, and 5.0 are documented
 
 ***
 
+[!INCLUDE [cryptography-apis-not-supported-on-blazor-webassembly](../../../includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md)]
```
Comment (scottaddie) [inline docs/core/compatibility/aspnetcore.md:197]:
```suggestion
[!INCLUDE[Cryptography APIs not supported on Blazor WebAssembly](~/includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md)]
```

File: docs/core/compatibility/cryptography.md [hunk 2]
```diff
@@ -18,6 +19,10 @@ The following breaking changes are documented on this page:
 
 ## .NET 5.0
 
+[!INCLUDE [cryptography-apis-not-supported-on-blazor-webassembly](../../../includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md)]
```
Comment (scottaddie) [inline docs/core/compatibility/cryptography.md:22]:
```suggestion
[!INCLUDE[Cryptography APIs not supported on Blazor WebAssembly](~/includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md)]
```

File: docs/core/compatibility/3.1-5.0.md [hunk 3]
```diff
@@ -263,8 +263,13 @@ If you're migrating from version 3.1 of .NET Core, ASP.NET Core, or EF Core to v
 
 ## Cryptography
 
+- [System.Security.Cryptography APIs not supported on Blazor WebAssembly](#systemsecuritycryptography-apis-not-supported-on-blazor-webassembly)
 - [System.Security.Cryptography.Oid is functionally init-only](#systemsecuritycryptographyoid-is-functionally-init-only)
 
+[!INCLUDE [cryptography-apis-not-supported-on-blazor-webassembly](../../../includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md)]
```
Comment (scottaddie) [inline docs/core/compatibility/3.1-5.0.md:269]:
```suggestion
[!INCLUDE[Cryptography APIs not supported on Blazor WebAssembly](~/includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md)]
```

File: docs/core/compatibility/aspnetcore.md (hunk 1 above)
Comment (gewarren) [reply docs/core/compatibility/aspnetcore.md:197]: I'll accept these suggestions, but just wondering if it actually makes any difference?

File: includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+### System.Security.Cryptography APIs not supported on Blazor WebAssembly
+
+<xref:System.Security.Cryptography> APIs throw a <xref:System.PlatformNotSupportedException> at run time on the `browser` platform.
+
+#### Change description
+
+In previous .NET versions, you can use the <xref:System.Security.Cryptography> APIs in the Blazor WebAssembly configuration. In .NET 5.0 and later versions, these APIs throw a <xref:System.PlatformNotSupportedException> when invoked from projects that use the `Microsoft.NET.Sdk.BlazorWebAssembly` SDK or that include `browser` as a supported platform in the project file.
```
Comment (scottaddie) [inline includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md:7]: @steveisok @pranavkm What's meant by "`browser` as a supported platform in the project file"? Does that refer to the following property in the project file of a Blazor WebAssembly project?

```xml
<RuntimeIdentifier>browser-wasm</RuntimeIdentifier>
```

Comment (pranavkm) [reply includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md:7]: https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-5-release-candidate-1/#browser-compatibility-analyzer-for-blazor-webassembly

File: includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+### System.Security.Cryptography APIs not supported on Blazor WebAssembly
+
+<xref:System.Security.Cryptography> APIs throw a <xref:System.PlatformNotSupportedException> at run time when run on a browser.
+
+#### Change description
+
+In previous .NET versions, the <xref:System.Security.Cryptography> APIs aren't available to Blazor WebAssembly apps. Starting in .NET 5.0, Blazor WebAssembly apps target the full .NET 5 API surface area, however, not all .NET 5 APIs are supported due to browser sandbox constraints. In .NET 5.0 and later versions, the <xref:System.Security.Cryptography> APIs throw a <xref:System.PlatformNotSupportedException> when running on WebAssembly.
```
Comment (gewarren) [inline includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md:7]: @steveisok Is the first sentence a correct statement? Or were the APIs available but just didn't throw the exception?

"In previous .NET versions, the <xref:System.Security.Cryptography> APIs aren't available to Blazor WebAssembly apps."

File: includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md (hunk 4 above)
Comment (scottaddie) [reply includes/core-changes/cryptography/5.0/cryptography-apis-not-supported-on-blazor-webassembly.md]: @gewarren I recommend clarifying this part by including `<SupportedPlatform Include="browser" />` (as mentioned in the linked blog post).

**Comment by gewarren** [comment]
Ping for review/approval so I can get this merged. Thanks!

**Review by CamSoper — APPROVED** [review]

**Pull request merged**